src/React/Basic.purs

```diff
@@ -369,6 +373,46 @@ foreign import toReactComponent
   -> { render :: Self props state -> JSX | spec }
   -> ReactComponent { | jsProps }
 
+type ReactContext a =
+  { provider :: ReactComponent { value :: a, children :: Array JSX }
+  , consumer :: ReactComponent { children :: a -> Array JSX }
+  }
+
+-- | Create a `ReactContext` given a default value. Use `provider` and `consumer`
+-- | to provide and consume context values. Alternatively, use the fields of
+-- | `ReactContext` directly if a `ReactComponent` is required for interop.
+-- |
+-- | ```purs
+-- | render self =
+-- |   R.div_
+-- |   [ R.button
+-- |     { onClick: capture_ $ self.setState \s -> s { counter = s.counter + 1 }
+-- |     , children: [ R.text "Tick!" ]
+-- |     }
+-- |   , provider countContext self.state.counter
+-- |     [ consumer countContext \counter ->
+-- |        [ R.text $ "Ticks: " <> (show counter)
+-- |        ]
+-- |     ]
+-- |   ]
+-- | ```
+-- |
+-- | __*See also:* `provider`, `consumer`, React's documentation regarding Context__
+foreign import createContext :: forall a. a -> ReactContext a
```

Comment:
This should have the type `EffectFn1 a (ReactContext a)`. I know we've cheated in other places but we're working to get away from that :sweat_smile:. Probably best to name this `createContext_`, hidden, and export `createContext = runEffectFn1 createContext_` instead. Alternatively the FFI could be curried manually and this could just have the type `a -> Effect (ReactContext a)`